### ISSUE TITLE
Automatic registeration of page layouts

### ISSUE THREAD

**Issue by MoOx** [MoOx/phenomic#676]
Easy!

``` js
import React, { Component } from "react"
import { Route } from "react-router"

import AppContainer from "../components/AppContainer"
import PhenomicPageContainer from "phenomic/lib/PageContainer"

/* ! HERE IS THE TRICK */
const reqLayouts = require.context("../layouts", true, /\/index\.js$/)
const layouts = reqLayouts.keys().reduce((layouts, layout) => ({
  ...layouts,
  [layout.replace("./", "").replace("/index.js", "")]:
    reqLayouts(layout).default,
}), {})
/* ! */

class PageContainer extends Component {
  render() {
    const { props } = this
    return (
      <PhenomicPageContainer
        { ...props }
        layouts={ layouts }
      />
    )
  }
}

export default (
  <Route component={ AppContainer }>
    <Route path="*" component={ PageContainer } />
  </Route>
)
```

We should add this in the default theme and also try to reduce this code, maybe using a function that accept the "reqLayouts". An issue might be the context that will be used for the require (same thing as the hot reload for markdown...)

**Comment by thangngoc89** [comment]
Can we force users to use `web_modules/layouts` as layouts location ? This would eliminate a lots of unneeded code in the boilerplate

**Comment by MoOx** [comment]
We could highly recommend, but nothing should be mandatory. Especially file tree, which is likely to be adjusted depending on the need, context, legacy etc...
But I am not even sure that with this restriction, the problem will go away. The require.context usage can be very tricky from my previous tests (for markdown hot loading).

**Comment by thangngoc89** [comment]
I used `require.context` a few times and like you said, it's very tricky.

**Comment by DavidWells** [comment]
I like the idea. =) 

I wanted this originally but realized that it might be limiting for people who want custom file directories.

Maybe if you make that trick an optional phenomic util people can use?

**Comment by bloodyowl** [comment]
Going to be fixed with #925 (moving to user-land)
